storage/src/journal/fixed.rs

```diff
@@ -268,36 +277,41 @@ impl<B: Blob, E: Storage<B>, A: Array> Journal<B, E, A> {
         }
 
         // Truncate the rewind blob to the correct offset.
-        let rewind_blob = match self.blobs.get_mut(&rewind_to_blob_index) {
+        let blob = match self.lru_cache.lock().await.pop(&rewind_to_blob_index) {
             Some(blob) => blob,
-            None => return Err(Error::MissingBlob(rewind_to_blob_index)),
+            None => {
+                let Ok(blob) = self
+                    .runtime
+                    .open(&self.cfg.partition, &rewind_to_blob_index.to_be_bytes())
+                    .await
+                else {
+                    return Err(Error::MissingBlob(rewind_to_blob_index));
+                };
+                blob
+            }
         };
+        self.set_newest_blob(rewind_to_blob_index, blob);
+        let (_, newest_blob) = self.newest_blob();
         let rewind_to_offset = (rewind_to_size % self.cfg.items_per_blob) * Self::CHUNK_SIZE as u64;
-        rewind_blob.truncate(rewind_to_offset).await?;
+        newest_blob.truncate(rewind_to_offset).await?;
         Ok(())
     }
 
     /// Read the item at the given position in the journal.
     pub async fn read(&self, item_position: u64) -> Result<A, Error> {
         let blob_index = item_position / self.cfg.items_per_blob;
-
-        let blob = match self.blobs.get(&blob_index) {
-            Some(blob) => blob,
-            None => {
-                let newest_blob = self.newest_blob();
-                if blob_index > newest_blob.0 {
-                    return Err(Error::InvalidItem(item_position));
-                }
-                assert!(blob_index < self.oldest_blob().0);
-                return Err(Error::ItemPruned(item_position));
-            }
-        };
-
-        let item_index = item_position % self.cfg.items_per_blob;
+        if blob_index > self.newest_index_blob.0 {
+            return Err(Error::InvalidItem(item_position));
+        }
+        if blob_index < self.min_blob_index {
+            return Err(Error::ItemPruned(item_position));
+        }
+        let items_per_blob = self.cfg.items_per_blob;
```

Comment:
since this is only used once I'd just remove the declaration and use self.cfg.items_per_blob directly on L311.